lib/mllp/sender.ex

```diff
@@ -1,10 +1,8 @@
 defmodule MLLP.SenderContract do
-  @type sender_error ::
-          :unspecified_error
-          | :reply_timeout_exceeded
-          | :not_connected
-          | :closed
-          | :tcp_error
+  @type error_type :: :send_error | :recv_error
+  @type error_reason :: :closed | :timeout | :inet.posix()
+
+  @type sender_error :: :not_connected | %{type: error_type(), reason: error_reason()}
```

Comment:
We discussed changing thee shape of the returned error tuples to this.  That leaves `:not_connecteed` as a different shape. I don't have strong feelings, but maybe you think the shape should be uniform?

Reply:
yes, we should make it consistent!

Reply:
Will adjust to return thee map with `:connect_failure` as the type.